lib/phoenix/endpoint/instrumenter.ex

```diff
@@ -0,0 +1,138 @@
+defmodule Phoenix.Endpoint.Instrument do
+  @doc false
+  defmacro __using__(_opts) do
+    quote unquote: false do
+      # The `otp_app` variable is set by Phoenix.Endpoint.
+      # TODO: maybe it's better to make Phoenix endpoints export __otp_app__()
+      # or something similar?
+      config        = Application.get_env(var!(otp_app), __MODULE__, [])
+      instrumenters = config[:instrumentation] || []
```

Comment:
Let's keep the current pattern. Most of the times we don't care about the distinction between those and this code snippet is more concise.